packages/apps/plugins/plugin-map/src/components/MapDetectLocations.ts

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { CollectionType, TableType, type MapType } from '@braneframe/types';
+import { getTypename, type DynamicSchema } from '@dxos/echo-schema';
+import { Filter, getSpace, useQuery } from '@dxos/react-client/echo';
+
+import { type Marker } from './MapControl';
+
+/**
+ * Find table rows that are in the same collection as the map.
+ * Return the ones with `latitude` and `longitude` fields,
+ * converted to `Marker`s.
+ *
+ * NOTE: This collects rows from all the collections the map is in,
+ * even if there's more than one. At the time of writing this behavior
+ * won't be exercised since being included in multiple collections
+ * is disabled in the UI.
+ *
+ * If we enable being in more than one collection again we should
+ * reconsider what behavior we actually want in that case.
+ */
+export const useMapDetectLocations = (map: MapType): Marker[] => {
+  const space = getSpace(map);
+
+  const parentCollections = useQuery<CollectionType>(
+    space,
+    [
+      Filter.schema(CollectionType),
+      (collection: any) => {
+        if (!(collection instanceof CollectionType)) {
+          return false;
+        }
+        return collection.objects.some((obj) => obj?.id === map.id);
+      },
+    ],
+    undefined,
+    [space],
+  );
+
+  const siblingIds = new Set(
+    parentCollections.flatMap((collection) => collection.objects.filter(hasId).map((obj) => obj.id)),
+  );
+
+  const siblingTables = useQuery<TableType>(
+    space,
+    [
+      Filter.schema(TableType),
+      (table: any) => siblingIds.has(table.id),
+      (table: any) => {
+        if (!(table instanceof TableType)) {
+          return false;
+        }
+        return table.schema ? schemaHasLatitudeAndLongitude(table.schema) : false;
+      },
+    ],
+    undefined,
+    [Array.from(siblingIds).sort().join('_')],
```

Comment:
If there's a more idiomatic way to deterministically stringify here, let me know

Reply:
I would call `JSON.stringify(Array.from(siblingsIds))` personally.

Reply:
Done! Doubly good you mentioned this because I was using `siblingIds` down below where I should have been using `schemaIds`. Now fixed.